src/components/Issues.js

```diff
@@ -1,42 +1,97 @@
 import styled from 'styled-components/macro';
 import pin from '../images/pin.svg';
+import pinFill from '../images/pin-fill.svg';
 
-function Issues({ issues }) {
+export default function Issues({ issues, togglePin }) {
   return (
     <>
-      {issues.map(issue => (
-        <Wrapper key={issue.id} title={issue.title} state={issue.state}>
-          <Title>{issue.title}</Title>
-          <State>{issue.state}</State>
-        </Wrapper>
-      ))}
+      <HeadingWrapper>
+        <HeadingTitle>Title</HeadingTitle>
+        <HeadingState>State</HeadingState>
+      </HeadingWrapper>
+
+      {issues.map(issue => {
+        const isPinned = issues.find(
+          issueId => issueId.id === issue.id
+        ).clicked;
+
+        return (
+          <Wrapper key={issue.id} title={issue.title} state={issue.state}>
+            <IssueTitle>{issue.title}</IssueTitle>
+            <IssueState>{issue.state}</IssueState>
+            <Button onClick={() => handleClick(issue.id)}>
+              {
+                <img
+                  id={issue.id}
+                  src={isPinned ? pinFill : pin}
+                  alt="pin"
+                  width="16"
+                  heigth="16"
+                />
+              }
+            </Button>
+          </Wrapper>
+        );
+      })}
     </>
   );
+
+  function handleClick(buttonId) {
+    togglePin(buttonId);
+  }
```

Comment:
isn´t it possible to give the tooglePin function directly to the button because the only thing the handleClick does is call another function?

Reply:
You are right, i can call direct the togglePin function from the button